models/_sector/nft/trades/ethereum/platforms/x2y2_ethereum_base_trades.sql

```diff
@@ -20,42 +21,42 @@ WITH src_evt_inventory as (
     ,evt_block_number as block_number
     ,evt_tx_hash as tx_hash
     ,contract_address as project_contract_address
-    ,case when intent = 1 then taker else maker end as buyer
-    ,case when intent = 1 then maker else taker end as seller
-    ,'0x' || substring(get_json_object(inv.item, '$.data'), 155, 40) as nft_contract_address
-    ,bytea2numeric_v3(substring(get_json_object(inv.item, '$.data'), 195,64)) as nft_token_id
-    ,CAST(1 AS DECIMAL(38,0)) AS nft_amount
-    ,case when intent = 1 then 'Buy' else 'Offer Accepted' end as trade_category
+    ,case when intent = cast(1 as uint256) then taker else maker end as buyer
+    ,case when intent = cast(1 as uint256) then maker else taker end as seller
+    ,from_hex('0x' || substring(JSON_EXTRACT_SCALAR(inv.item, '$.data'), 155, 40)) as nft_contract_address
```

Comment:
nit: it's not necessary to prepend the '0x'. `from_hex` accepts both `0x`-prefixed and unprefixed byte arrays.
```suggestion
    ,from_hex(substring(JSON_EXTRACT_SCALAR(inv.item, '$.data'), 155, 40)) as nft_contract_address
```